nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/test/java/org/nuxeo/ecm/platform/audit/io/LogEntryCSVWriterTest.java

```diff
@@ -0,0 +1,135 @@
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+@Features(AuditFeature.class)
+@Deploy("org.nuxeo.ecm.platform.audit.tests:test-pageprovider-contrib.xml")
+public class LogEntryCSVWriterTest extends AbstractCSVWriterTest.External<LogEntryCSVWriter, LogEntry> {
+
+    @Inject
+    private PageProviderService pps;
+
+    @Inject
+    private CoreSession session;
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    public LogEntryCSVWriterTest() {
+        super(LogEntryCSVWriter.class, LogEntry.class);
+    }
+
+    @Test
+    public void testDefaultProperties() throws IOException {
+        DocumentModel file = session.createDocumentModel("/", "myFile", "File");
+        file = session.createDocument(file);
+        // LogEntry are event driven. They are created asynchronously
+        txFeature.nextTransaction();
+        PageProvider<?> pp = pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", null, Long.valueOf(20), Long.valueOf(0),
+                new HashMap<>(), file);
+        @SuppressWarnings("unchecked")
+        List<LogEntry> entries = (List<LogEntry>) pp.getCurrentPage();
+        if (entries.isEmpty()) {
+            // if there is nothing to test, the test would always succeed.
+            fail();
+        }
+        // No fetch param in the context, retrieve all default properties
+        RenderingContext renderingCtx = RenderingContext.CtxBuilder.get();
+        for (LogEntry entry : entries) {
+            CSVAssert csv = csvAssert(entry, renderingCtx);
+            csv.has("id");
+            csv.has("category").isEquals("eventDocumentCategory");
+            csv.has("principalName").isEquals("Administrator"); // NOSONAR
+            csv.has("comment").isEquals("");
+            csv.has("docLifeCycle").isEquals("project"); // NOSONAR
+            csv.has("docPath").isEquals("/myFile"); // NOSONAR
+            csv.has("docType").isEquals("File"); // NOSONAR
+            csv.has("docUUID").isEquals(file.getId());
+            csv.has("eventId").isEquals(DocumentEventTypes.DOCUMENT_CREATED);
+            csv.has("repositoryId").isEquals("test");
+            csv.has("eventDate");
+            csv.has("logDate");
+        }
+    }
+
+    @Test
+    public void testCustomProperties() throws IOException {
+        DocumentModel file = session.createDocumentModel("/", "myFile", "File");
+        file = session.createDocument(file);
+        // LogEntry are event driven. They are created asynchronously
+        txFeature.nextTransaction();
+        PageProvider<?> pp = pps.getPageProvider("DOCUMENT_HISTORY_PROVIDER", null, Long.valueOf(20), Long.valueOf(0),
+                new HashMap<>(), file);
+        @SuppressWarnings("unchecked")
+        List<LogEntry> entries = (List<LogEntry>) pp.getCurrentPage();
+        if (entries.isEmpty()) {
+            // if there is nothing to test, the test would always succeed.
+            fail();
+        }
```

Comment:
Could be factorized in a method returning `List<LogEntry>`, adding a `protected DocumentModel file` member.